client/src/helpers/nova/novaTimeUtils.ts

```diff
@@ -0,0 +1,58 @@
+import { ProtocolInfo } from "@iota/sdk-wasm-nova/web";
+
+export function unixTimestampToSlotIndexConverter(protocolInfo: ProtocolInfo): (unixTimestampSeconds: number) => number {
+    return (unixTimestampSeconds: number) => {
+        const genesisSlot = protocolInfo.parameters.genesisSlot;
+        const genesisUnixTimestamp = protocolInfo.parameters.genesisUnixTimestamp;
+        const slotDurationInSeconds = protocolInfo.parameters.slotDurationInSeconds;
+
+        const elapsedTime = unixTimestampSeconds - Number(genesisUnixTimestamp);
+
+        if (elapsedTime < 0) {
+            return genesisSlot;
+        }
+
+        return genesisSlot + Math.floor(elapsedTime / slotDurationInSeconds) + 1;
```

Comment:
just to im on the same page, why +1?

Reply:
> Otherwise the slot index is Genesis Slot + Elapsed Time/Slot Duration In Seconds + 1.
> Note: The + 1 is required because slots are counted starting from 1. 0 is reserved for times before the genesis (if Genesis Slot = 0), which has to be addressable as its own slot.
From https://github.com/iotaledger/tips/blob/tip46/tips/TIP-0046/tip-0046.md#time